src/Database/Validator/Index.php

```diff
@@ -25,39 +34,128 @@ public function getDescription(): string
         return $this->message;
     }
 
+    /**
+     * @param Document $collection
+     * @return bool
+     */
+    public function checkEmptyIndexAttributes(Document $collection): bool
+    {
+        foreach ($collection->getAttribute('indexes', []) as $index) {
+            if (empty($index->getAttribute('attributes', []))) {
+                $this->message = 'No attributes provided for index';
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param Document $collection
+     * @return bool
+     * @throws DatabaseException
+     */
+    public function checkFulltextIndexNonString(Document $collection): bool
+    {
+        foreach ($collection->getAttribute('indexes', []) as $index) {
+            if ($index->getAttribute('type') === Database::INDEX_FULLTEXT) {
+                foreach ($index->getAttribute('attributes', []) as $attributeName) {
+                    $attribute = $this->attributes[$attributeName] ?? new Document([]);
+                    if ($attribute->getAttribute('type', '') !== Database::VAR_STRING) {
+                        $this->message = 'Attribute "'.$attribute->getAttribute('key', $attribute->getAttribute('$id')).'" cannot be part of a FULLTEXT index';
+                        return false;
+                    }
+                }
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * @param Document $collection
+     * @return bool
+     */
+    public function checkIndexLength(Document $collection): bool
+    {
+        foreach ($collection->getAttribute('indexes', []) as $index) {
+            if ($index->getAttribute('type') === Database::INDEX_FULLTEXT) {
+                return true;
+            }
+
+            $total = 0;
+            $lengths = $index->getAttribute('lengths', []);
+
+            foreach ($index->getAttribute('attributes', []) as $attributePosition => $attributeName) {
+                $attribute = $this->attributes[$attributeName];
+
+                switch ($attribute->getAttribute('type')) {
+                    case Database::VAR_STRING:
+                        $attributeSize = $attribute->getAttribute('size', 0);
+                        $indexLength = $lengths[$attributePosition] ?? $attributeSize;
+                        break;
+
+                    case Database::VAR_FLOAT:
+                        $attributeSize = 2; // 8 bytes / 4 mb4
+                        $indexLength = 2;
+                        break;
+
+                    default:
+                        $attributeSize = 1; // 4 bytes / 4 mb4
+                        $indexLength = 1;
+                        break;
+                }
+
+                if ($indexLength > $attributeSize) {
+                    $this->message = 'Index length("'.$indexLength.'") is longer than the key part for "'.$attributeName.'("'.$attributeSize.'")"';
```

Comment:
Let's remove the parentheses here:

```suggestion
                    $this->message = 'Index length '.$indexLength.' is larger than the size for '.$attributeName.': '.$attributeSize.'"';
```